### ISSUE TITLE
Typescript compilation error regarding a bignumber.js import

### ISSUE THREAD

**Issue by dapplion** [ethereum/web3.js#3386]
#### Expected behavior

Build web3 with typescript without errors

#### Actual behavior

```
node_modules/web3-core/types/index.d.ts(31,23): error TS2307: Cannot find module 'bignumber.js'.
node_modules/web3-eth/types/index.d.ts(51,25): error TS2307: Cannot find module 'bignumber.js'.
```

see this Github Action to reproduce, logs, and configuration 
https://github.com/dapplion/web3-tsc-test/runs/461815145

**Comment by cgewecke** [comment]
@dapplion Thanks for reporting. Have opened #3390 to resolve this. 

An interim fix is to install bignumber.js as a dependency in your project.

**Comment by dapplion** [comment]
Thanks for pushing a fix fast. Adding `bignumber.js` as a direct dependency also fixed the problem in my case.

**Comment by holgerd77** [comment]
Fixed by #3390

**Comment by cgewecke** [comment]
Re-opening for visibility until `1.2.7` is published.
